make/image.mk

```diff
@@ -0,0 +1,110 @@
+		  -v ${PWD}:/workspace
+
+%:
+ifeq ($(MAKE_IN_DOCKER), 1)
+# Build builder image
+	@$(DOCKER_BUILD_CMD) -t $(DOCKER_BUILDER_IMAGE):$(DOCKER_BUILDER_TAG) - < ../tools/docker/Dockerfile
+# Run make inside the builder image
+# Run with MAKE_IN_DOCKER=0 to eliminate an infinite loop
+	@$(DOCKER_RUN_CMD) $(DOCKER_BUILDER_IMAGE):$(DOCKER_BUILDER_TAG) MAKE_IN_DOCKER=0 $@
+endif
+
+# Docker variables
+# REGISTRY is the image registry to use for build and push image targets.
+REGISTRY ?= docker.io/envoyproxy
+# IMAGE is the image URL for build and push image targets.
+IMAGE ?= ${REGISTRY}/gateway-dev
+# Tag is the tag to use for build and push image targets.
+TAG ?= $(REV)
+
+DOCKER := docker
+DOCKER_SUPPORTED_API_VERSION ?= 1.32
+
+ifdef HTTP_PROXY
+_DOCKER_BUILD_EXTRA_ARGS += --build-arg HTTP_PROXY=${HTTP_PROXY}
+endif
+
+ifneq ($(EXTRA_ARGS), )
+_DOCKER_BUILD_EXTRA_ARGS += $(EXTRA_ARGS)
+endif
+
+# Determine image files by looking into build/docker/*/Dockerfile
+IMAGES_DIR ?= $(wildcard ${ROOT_DIR}/docker/*)
+# Determine images names by stripping out the dir names
+IMAGES ?= envoy-gateway
+IMAGE_PLATFORMS ?= linux_amd64 linux_arm64 
+
+ifeq (${IMAGES},)
+  $(error Could not determine IMAGES, set ROOT_DIR or run in source dir)
+endif
+
+.PHONY: image.verify
+image.verify:
+	$(eval API_VERSION := $(shell $(DOCKER) version | grep -E 'API version: {1,6}[0-9]' | head -n1 | awk '{print $$3} END { if (NR==0) print 0}' ))
+	$(eval PASS := $(shell echo "$(API_VERSION) > $(DOCKER_SUPPORTED_API_VERSION)" | bc))
+	@if [ $(PASS) -ne 1 ]; then \
+		$(DOCKER) -v ;\
+		echo "Unsupported docker version. Docker API version should be greater than $(DOCKER_SUPPORTED_API_VERSION)"; \
+		exit 1; \
+	fi
+
+.PHONY: image.build
+image.build: image.verify  $(addprefix image.build., $(addprefix $(IMAGE_PLAT)., $(IMAGES)))
+
+.PHONY: image.build.multiarch
+image.build.multiarch: image.verify  $(foreach p,$(IMAGE_PLATFORMS),$(addprefix image.build., $(addprefix $(p)., $(IMAGES))))
+
+.PHONY: image.build.%
+image.build.%: go.build.%
+	$(eval IMAGE := $(COMMAND))
+	$(eval IMAGE_PLAT := $(subst _,/,$(PLATFORM)))
+	@echo "===========> Building image $(IMAGE) in tag $(TAG) for $(IMAGE_PLAT)"
+	@mkdir -p $(TMP_DIR)/$(IMAGE)
+	@cat $(ROOT_DIR)/docker/$(IMAGE)/Dockerfile\
+		>$(TMP_DIR)/$(IMAGE)/Dockerfile
+	@cp $(OUTPUT_DIR)/$(IMAGE_PLAT)/$(IMAGE) $(TMP_DIR)/$(IMAGE)/
+	$(eval BUILD_SUFFIX := $(_DOCKER_BUILD_EXTRA_ARGS) --pull -t $(REGISTRY)/$(IMAGE):$(TAG) $(TMP_DIR)/$(IMAGE))
+	$(eval BUILD_SUFFIX_ARM := $(_DOCKER_BUILD_EXTRA_ARGS) --pull -t $(REGISTRY)/$(IMAGE).$(ARCH):$(TAG) $(TMP_DIR)/$(IMAGE))
+	@if [ "$(ARCH)" == "amd64" ]; then \
+		echo "===========> Creating image tag $(REGISTRY)/$(IMAGE):$(TAG) for $(ARCH)"; \
+		$(DOCKER) build --platform $(IMAGE_PLAT) $(BUILD_SUFFIX); \
+	else \
+		echo "===========> Creating image tag $(REGISTRY)/$(IMAGE).$(ARCH):$(TAG) for $(ARCH)"; \
+		$(DOCKER) build --platform $(IMAGE_PLAT) $(BUILD_SUFFIX_ARM); \
```

Comment:
we shouldnt have a separate suffix for arm, either a single arch image for the user or multi arch similar to what CI creates https://hub.docker.com/r/envoyproxy/gateway-dev/tags (multi arch manifest)

Reply:
Looks good, thanks, this suffix looks weird for me too.

Reply:
Removed this.